### ISSUE TITLE
Unpermitted parameters when hive (re-)registers

### ISSUE THREAD

**Issue by davidbuckhurst** [bbc/hive_mind#107]
When a hive registers, I see the following in the logs ... yet everything appears to be working:

Unpermitted parameters: brand, device_type, hostname, model, operating_system_name, operating_system_version, runner_plugins, version

**Comment by jrmhaig** [comment]
Check to see if this is still an issue.

**Comment by jrmhaig** [comment]
Yes, this is still an issue but not a major problem. This warning comes from the following line at the top of the `register` method in `app/controllers/api/devices_controller.rb`:

```
    create_parameters = device_params
```

The parameters that are passed include parameters that are specific for the device type but `device_params` only validates the fields that are related to the core Device model. I.e., the parameters "brand, device_type, hostname, model, operating_system_name, operating_system_version, runner_plugins, version" are relevant to `HiveMindHive::Plugin`.

To avoid this warning `device_params` needs to be more clever, taking into account `device_params[:device_type]`.

**Comment by jrmhaig** [comment]
I have done some work on this in this branch:

https://github.com/bbc/hive_mind/tree/fix_unpermitted_parameters

but it is not yet complete.
